httplogs/httplog.go

```diff
@@ -0,0 +1,148 @@
+package httplogs
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+
+	"strconv"
+	"time"
+
+	objConsulAgent "github.com/carwale/golibraries/consulagent"
+	"github.com/carwale/golibraries/gologger"
+)
+
+var _gLogConfig *GlobalParameters
+
+// GlobalParameters is the class used to store global variables
+type GlobalParameters struct {
+	consulAgent            *objConsulAgent.ConsulAgent
+	serviceLogger          *gologger.CustomLogger
+	serviceName            string
+	consulIP               string
+	isMonitoringLogEnabled bool
+}
+
+// Options sets a variable of GlobalParameters
+type Options func(lb *GlobalParameters)
+
+// HTTPAccessLoggingWrapper is wrapper to neable access logs
+func HTTPAccessLoggingWrapper(h http.Handler) http.Handler {
+	loggingFn := func(w http.ResponseWriter, r *http.Request) {
+		lrw := httploggingResponseWriter{
+			ResponseWriter: w,
+			rData: &responseData{
+				status: 0,
+				size:   0,
+			},
+		}
+
+		h.ServeHTTP(&lrw, r) // inject our implementation of http.ResponseWriter
+		logHTTPLogs(r, lrw.rData.status, lrw.rData.size)
+	}
+	return http.HandlerFunc(loggingFn)
+}
+
+// InitLogging acts as a constructor to initialize the logging service and
+// initailize the struct
+func InitLogging(serviceName string, options ...Options) {
+	_gLogConfig = setDefaultConfig(serviceName)
+	for _, option := range options {
+		option(_gLogConfig)
+	}
+	setBasicConfig(serviceName)
+}
+
+// SetLogger (mandatory) parameter in order to configure logger
+func SetLogger(customLogger *gologger.CustomLogger) Options {
+	return func(lb *GlobalParameters) { lb.serviceLogger = customLogger }
+}
+
+// SetConsulIP (mandatory) default value is localhost, thus service has to change the
+// consul ip based on environment
+func SetConsulIP(consultIP string) Options {
+	return func(al *GlobalParameters) { al.consulIP = consultIP }
+}
+
+func setDefaultConfig(serviceName string) *GlobalParameters {
+	return &GlobalParameters{
+		consulIP:      "127.0.0.1:8500",
+		serviceLogger: gologger.NewLogger(),
+		serviceName:   serviceName,
+	}
+}
+
+// SetBasicConfig start point of the request
+func setBasicConfig(serviceName string) {
+	_gLogConfig.consulAgent = objConsulAgent.NewConsulAgent(
+		objConsulAgent.ConsulHost(_gLogConfig.consulIP),
+		objConsulAgent.Logger(_gLogConfig.serviceLogger),
+	)
+
+	monitoringKey := getMonitoringKey(serviceName)
+	go checkHTTPLogStatus(monitoringKey)
+}
+
+// infinite loop checking the key 'access_logs'
+func checkHTTPLogStatus(key string) {
+	for {
+		_gLogConfig.serviceLogger.LogDebug("The value of access log for " + _gLogConfig.serviceName + " is:" + strconv.FormatBool(_gLogConfig.isMonitoringLogEnabled))
+		time.Sleep(10 * time.Second)
```

Comment:
this time was added for testing purpose. Will update it to 15minutes once branch is approved